sdw_util/Util.py

```diff
@@ -148,6 +150,45 @@ def get_qubes_version():
     return version
 
 
+def get_whonix_version():
+    """
+    Obtain Whonix version from /srv/formulas/base/virtual-machine-formula/qvm/whonix.jinja
+
+    In '.sls' files the whonix version is obtained with
+
+        {% import "qvm/whonix.jinja" as whonix %}
+        {{ whonix.whonix_version }}
+
+    This function extracts this value so that it can be on this script.
+    """
+
+    # "caching" because qubesctl takes several seconds
+    global whonix_version  # noqa: PLW0603
+    if whonix_version:
+        return whonix_version
+
+    try:
+        output = subprocess.check_output(
+            [
+                "sudo",
+                "qubesctl",
+                "jinja.load_map",
+                "qvm/whonix.jinja",
+                "whonix_version",
+                "--out",
+                "newline_values_only",
+            ],  # avoid "local:" prefix
+            universal_newlines=True,
+        )
+        whonix_version = int(output.strip())
+    except subprocess.CalledProcessError:
```

Comment:
Is this the right behavior? If we can't get the whonix version, shouldn't we hard fail? (None of the callers have any error handling to check for a `None` value, we're just going to end up with operations on `whonix-gateway-None`.